### ISSUE TITLE
Spontaneous crashes during session caused by backup saving

### ISSUE THREAD

**Issue by micahwoodard** [AllenNeuralDynamics/dynamic-foraging-task#1174]
RA's have been having issues where gui's will crash mid session. It seems like this may be related to a bug in the save thread. Some examples of these instances are in issues: [#800](https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/800), ⁠[#797](https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/797), ⁠[#791](https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/791), ⁠[#788](https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/788), ⁠[#779](https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/779), [#752](https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/752)

**Comment by XX-Yin** [comment]
I also speculate that it’s related to the backup save. The interference between different threads may cause crash, but the exact reason is unknown. I will turn off the backup save today through changing the settings in all behavior rig.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Micah Woodard ***@***.***>
Sent: Friday, December 13, 2024 8:31:13 AM
To: AllenNeuralDynamics/dynamic-foraging-task ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [AllenNeuralDynamics/dynamic-foraging-task] Spontaneous crashes during session (Issue #1174)


RA's have been having issues where gui's will crash mid session. It seems like this may be related to a bug in the save thread. Some examples of these instances are in issues: #800<https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/800>, ⁠#797<https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/797>, ⁠#791<https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/791>, ⁠#788<https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/788>, ⁠#779<https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/779>, #752<https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/752>

**Comment by micahwoodard** [comment]
@XX-Yin just to clarify, turning off backup save requires changing the setting.json file?

**Comment by XX-Yin** [comment]
Yes. I can do that for all behavior rig this afternoon

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by micahwoodard** [comment]
Is it something I could go do quickly now? I would be nice to be able to see if crashes were any better today

**Comment by XX-Yin** [comment]
I’m out this morning. It’s something like the save_each_trial. Can you check in the code?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by micahwoodard** [comment]
This is an example settings.json from 8B. I don't see anything related to saving. Is it in a different file?

 
[ForagingSettings.json](https://github.com/user-attachments/files/18131077/ForagingSettings.json)

**Comment by micahwoodard** [comment]
Oh it looks like if that parameter isn't found in the settings json it defaults to True. I can add it as false to rigs

**Comment by XX-Yin** [comment]
That’s right. Thanks

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by micahwoodard** [comment]
@XX-Yin for all of them? Is there any risks or downsides to that?

**Comment by XX-Yin** [comment]
@micahwoodard Yes, it's worth trying all of them. The only risk is that if the GUI crashes, the data won't be saved. However, the crashes seem to be caused by the backups themselves, the other stuff is pretty robust.

**Comment by micahwoodard** [comment]
ran into @bwtan-allen as I was updating settings files and he mentioned that it might be worth discussing with @hagikent or others who might be affected if session crash leads to loss of all data

**Comment by XX-Yin** [comment]
I suggest we do it (updating settings files) at least one week as it's most likely the crash is caused by the backup saving.

**Comment by micahwoodard** [comment]
Seems like switching save_every_trial to False has helped with crashing. @XX-Yin any idea where in the saving threading things could be going wrong?

**Comment by alexpiet** [comment]
@XX-Yin bumping this conversation. We should strive to get backup saving working again

**Comment by XX-Yin** [comment]
that makes sesnse

**Comment by alexpiet** [comment]
@JeremiahYCohen @ellahiltonvano This is the issue about the backup saving. It should be the highest priority to get this working again

**Comment by JeremiahYCohen** [comment]
Do we have a plan to implement this @XX-Yin @hagikent ?

**Comment by XX-Yin** [comment]
> Do we have a plan to implement this [@XX-Yin](https://github.com/XX-Yin) [@hagikent](https://github.com/hagikent) ?

@JeremiahYCohen I will try to fix this issue.

**Comment by hagikent** [comment]
this is an issue with the GUI.
FP data will be usually saved even upon crash (with some Nans).

**Comment by JeremiahYCohen** [comment]
@XX-Yin do we have updates here?

**Comment by XX-Yin** [comment]
@JeremiahYCohen I made a PR here https://github.com/AllenNeuralDynamics/dynamic-foraging-task/pull/1331. Waiting for review and merge.

**Comment by ellahiltonvano** [comment]
Closing since Xinxin's PR has been merged to main.

**Comment by micahwoodard** [comment]
@XX-Yin have the setting jsons in 446 and 447 been changed to add back in saving for each trial?

**Comment by ellahiltonvano** [comment]
@alexpiet can confirm that I changed them in 446 yesterday.
